lib/mail/fields/content_disposition_field.rb

```diff
@@ -9,6 +10,7 @@ class ContentDispositionField < StructuredField
 
     def initialize(value = nil, charset = 'utf-8')
       self.charset = charset
+      value = value.dup if value.is_a?(String) && value.frozen?
       ensure_filename_quoted(value)
       super(CAPITALIZED_FIELD, strip_field(FIELD_NAME, value), charset)
       self.parse
```

Comment:
I'm fine with changing that if you're ok with me changing that behavior ?- it is a private method so it's not public API